### PR TITLE
[bug-preprocessing] Fix phase-shift dtype

### PR DESCRIPTION
**Pull request by alejoe91**
and a typo in phy extractor

fixs #1125

### CODE REVIEW

File: spikeinterface/preprocessing/phase_shift.py [hunk 1]
```diff
@@ -49,15 +49,17 @@ def __init__(self, recording, margin_ms=40.,  inter_sample_shift=None, dtype=Non
...
-            if recording.get_dtype().kind in ('i', 'u'):
-                # because of the tapper on margin we need to force the computation to float32
-                # and then go back to the original buffer unless dtype is explicitly forced
-                dtype = np.dtype('float32')
-                force_dtype_back = recording.get_dtype()
- 
+            dtype = recording.get_dtype()
+        # because of the tapper on margin we need to force the computation to float32
+        # and then go back to the original buffer unless dtype is explicitly forced
+        tmp_dtype = np.dtype('float32') if dtype.kind in ('i', 'u') else dtype
```
Comment (samuelgarcia) [inline spikeinterface/preprocessing/phase_shift.py:55]: It is better to to put None instead dtype so it avoid the astype() in the get_trace()

**Comment by samuelgarcia** [comment]
Lets discuss this with a call.
Even it if fix a bug this PR force to many useless conversion I think.

File: spikeinterface/preprocessing/phase_shift.py [hunk 2]
```diff
@@ -47,17 +47,19 @@ def __init__(self, recording, margin_ms=40.,  inter_sample_shift=None, dtype=Non
...
-            if recording.get_dtype().kind in ('i', 'u'):
-                # because of the tapper on margin we need to force the computation to float32
-                # and then go back to the original buffer unless dtype is explicitly forced
-                dtype = np.dtype('float32')
-                force_dtype_back = recording.get_dtype()
- 
+            dtype = recording.get_dtype()
+        # the "apply_shift" function returns a float64 buffer. In case the dtype is different
+        # than float64, we need a temporary casting and to force the buffer back to the original dtype
+        if dtype.str != "<f8":
```
Comment (samuelgarcia) [inline spikeinterface/preprocessing/phase_shift.py:54]:
```suggestion
        if dtype != "float64":
```

File: spikeinterface/preprocessing/phase_shift.py [hunk 3]
```diff
@@ -47,17 +47,19 @@ def __init__(self, recording, margin_ms=40.,  inter_sample_shift=None, dtype=Non
...
+            tmp_dtype = np.dtype('float32')
```
Comment (samuelgarcia) [inline spikeinterface/preprocessing/phase_shift.py:55]:
```suggestion
            tmp_dtype = np.dtype('float64')
```

Comment (samuelgarcia) [reply spikeinterface/preprocessing/phase_shift.py:55]: apparently numpy fft is always float64. So lets directly move to float64 to avid 2 conversions.

File: spikeinterface/preprocessing/phase_shift.py (hunk 2 above)
Comment (samuelgarcia) [reply spikeinterface/preprocessing/phase_shift.py:54]: Here you were forcing the endianness.

File: spikeinterface/preprocessing/phase_shift.py (hunk 3 above)
Comment (alejoe91) [reply spikeinterface/preprocessing/phase_shift.py]: done

File: spikeinterface/preprocessing/phase_shift.py (hunk 2 above)
Comment (alejoe91) [reply spikeinterface/preprocessing/phase_shift.py]: done (used `str(dtype)` because dtype could be a np object)

**Comment by alejoe91** [comment]
@samuelgarcia ready to merge

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**